### ISSUE TITLE
Dynamic param values based on component properties

### ISSUE THREAD

**Issue by strzibny** [allmarkedup/lookbook#109]
Lookbook sounds like a great project!

I am currently evaluating if we can move on from [storybook-viewcomponent](https://github.com/jonspalmer/view_component_storybook) and one thing we use a lot is passing param values as a Ruby constant, e.g.:

```ruby
story :button do
   constructor(
      ..
      variant: select(ButtonComponent::VARIANTS, :secondary),
      ..
   )
end
```
The same has to be done with copying & pasting in Lookbook, e.g.:

```ruby
# @param variant [Symbol] select [primary, secondary]
```
And I was thinking if it would be possible to evaluate Ruby for the values, e.g.:

```ruby
# @param variant [Symbol] select `ButtonComponent::VARIANTS`
```
Is this possible?

**Comment by strzibny** [comment]
Okay, I went ahead and tried it. It's possible.
![lookbook_patch](https://user-images.githubusercontent.com/1395527/176871833-9a8c665e-f129-4ad9-877b-150186257d92.png)

So there should be a way to write a nicer patch and include it.

**Comment by allmarkedup** [comment]
Hey @strzibny - yes the current static nature of the `@params` tag arguments can be a bit limiting at times unfortunately. Addressing that by some sort of string evaluation is definitely _possible_ - [this issue](https://github.com/allmarkedup/lookbook/issues/104) is about potentially using ERB in comments which would just be just a slightly different syntax to achieve basically the same end result.

Up until now I've tried to steer away from any form of `eval`ing of Ruby code just to try to avoid any obvious security gotchas. But it's something I'm actively looking into as I realise that it would be very useful to people.

**Comment by strzibny** [comment]
I think ERB sounds good too.

**Comment by allmarkedup** [comment]
Hey @strzibny, I've just shipped v1.1.0 that includes support for dynamically generating options (either via a method in the preview class itself or through arbitrary code evaluation).

Full details are [in the docs](https://lookbook.build/guide/previews/params/#param-options) - I'm going to close this issue down now but let me know if you run into any issues if you get a chance to try it out :-)
